tools/building.py

```diff
@@ -48,18 +48,6 @@
 user_requested_exports = set()
 
 
-class ObjectFileInfo:
-  def __init__(self, returncode, output, defs=set(), undefs=set(), commons=set()):
-    self.returncode = returncode
-    self.output = output
-    self.defs = defs
-    self.undefs = undefs
-    self.commons = commons
```

Comment:
The only reason we used a class like this was to work around a Python multiprocessing pickling issue in circa 2014, where a basic dict would not pickle over properly to the subprocesses. Now that we don't use Python multiprocessing on Windows anymore, we can simplify to just do more flexible dictionaries.

Reply:
Not sure about that - we don't use namedtuples generally throughout the codebase, and attempting to access nonexisting fields will not work in either case. It feels more nonidiomatic than the current pattern.

Reply:
I have started using `namedtuple` in a few places (see`tools/webassembly.py`).   Also `Enum`.   I have found named tuple very nice and succinct way give more meaning and readability over plain old dictionaries and tuples.   

I'm not sure i understand what you are getting at with nonexistant fields?   That won't work on plain dictionary either will it?

I don't feel paricularly strong since the use of the data type is very local.  lgtm either way.